src/operators_dense.jl

```diff
@@ -108,7 +108,7 @@ function *(psi::Bra{BL,T}, op::AbstractOperator{BL,BR}) where {BL,BR,T}
     mul!(result,psi,op)
     return result
 end
-
+_parent = parent∘parent∘parent
```

Comment:
It occurs to me we could do `_parent(x) = (x === parent(x) ? x : _parent(parent(x)))`. I think maybe this is what you were getting at when we discussed?

Reply:
PS: If you make this change, please rebase on the master branch to get the latest CI config.

Reply:
I am not sure this will be type-stable. May be worth checking. Probably could replace `x === parent(x)` with `typeof(x) == typeof(parent(x))`.

Reply:
Maybe you meant 
```julia
_parent(x::T, x_parent::T) = x
_parent(x, x_parent) = _parent(x_parent, parent(x_parent))
_parent(x) = _parent(x, parent(x))
```
which ought to be type stable :)

Reply:
Kinda, but your implementation has the added benefit of actually working!

Reply:
Nice, I saw the original version of this in the PR and thought about a better solution. Came up with the current version too.